base/compiler/typeinfer.jl

```diff
@@ -759,15 +766,32 @@ function typeinf_edge(interp::AbstractInterpreter, method::Method, @nospecialize
         end
         typeinf(interp, frame)
         update_valid_age!(frame, caller)
-        return frame.bestguess, frame.inferred ? mi : nothing
+        bestguess = frame.bestguess
+        if isa(bestguess, Conditional)
```

Comment:
I'd rather split this out into a descriptive function e.g. `bestguess_to_interprocedural` that gets calls unconditionally and then checks this situation internally.

Reply:
Yeah good idea, I put `bestguess_to_interprocedural` into `finish(::InferenceState, ::AbstractInterpreter)`.